server/boot/controller-boot.js

```diff
@@ -0,0 +1,30 @@
+var controller;
+
+module.exports = function(app) {
+  // here app.controller(ctrl) creates a controller wrapper to register
+  // the 'ctrl' class with strong-remoting
+  app.controller(ctrl).remoteMethod('speak', {
+    http: { path: '/', verb: 'get' },
+    returns: { arg: 'msg', type: 'string', root: true },
+  });
+};
+
+function ctrl(app, ctx) {
+  // actually ends up registering at /api/ctrl
+  this.name = 'TestName';
```

Comment:
I thought it should have been registered at `"TestName"` but now I realise that I set the name to [`ctor.name`](https://github.com/strongloop/loopback/pull/2284/files#diff-5372f626ee15242f1e2c6eb31655b4faR593) which is the function name. `"TestName"` is actually the name of the instance.

@bajtos, do you have any ideas on how to set this? We can set the name outside of the constructor `ctrl.name = 'MyControllerName'` but that seems less than ideal to me. A swagger  spec file could take care of the issue when that happen but to use controllers without swagger we need another solution.